code/modules/projectiles/guns/specialist.dm

```diff
@@ -1296,3 +1296,95 @@
 	. = ..()
 	if(last_signal_flare_name)
 		. += SPAN_NOTICE("The last signal flare fired has the designation: [last_signal_flare_name]")
+
+//-------------------------------------------------------
+//P9 Sonic Harpoon Artillery Remote Projectile(SHARP) Rifle
+
+/obj/item/weapon/gun/rifle/sharp
+	name = "\improper P9 SHARP Rifle"
+	desc = "A experimental harpoon launcher rifle manufactured by Armat Systems. It's specialized for specific ammo types out of a 10-round magainze, best used for area denial and disruption."
+	icon_state = "sharprifle"
+	item_state = "sharp"
+	fire_sound = 'sound/weapons/gun_sharp.ogg' // to fix
+	reload_sound = 'sound/weapons/handling/m41_reload.ogg' // to fix
+	unload_sound = 'sound/weapons/handling/m41_unload.ogg' // to fix
+	unacidable = TRUE
+	indestructible = 1
+	muzzle_flash = null
+
+	current_mag = /obj/item/ammo_magazine/rifle/sharp/explosive
+	attachable_allowed = list(/obj/item/attachable/magnetic_harness)
+	matter = list("metal" = 6000)
```

Comment:
you know this means you can put it into an autolathe

Reply:
why do all the other guns have it